cli/flox-rust-sdk/src/models/environment/fetcher.rs

```diff
@@ -87,6 +90,29 @@ impl IncludeFetcher {
         Ok((manifest, name))
     }
 
+    /// Fetch a remote environment.
+    fn fetch_remote(
+        &self,
+        flox: &Flox,
+        remote: &EnvironmentRef,
+        name: &Option<String>,
+    ) -> Result<(Manifest, String), EnvironmentError> {
+        // Don't poison the default cache_dir.
+        let tempdir =
+            tempfile::tempdir_in(&flox.temp_dir).map_err(EnvironmentError::CreateTempDir)?;
+
+        let pointer =
+            ManagedPointer::new(remote.owner().clone(), remote.name().clone(), &flox.floxhub);
+        let environment = RemoteEnvironment::new_in(flox, tempdir.path(), pointer)?;
+
+        let manifest = environment.manifest(flox)?;
+        let name = name
```

Comment:
> Incidentally I'm growing less sure that https://github.com/flox/flox/commit/6828a500a7626f0463e7d3d1b2ad206f09218698 was a good idea

I'm curious how come?

Reply:
Most of the time I just want a `Lockfile` and don't care whether it's newly locked or not. A small percentage of the time I want `lockfile_if_up_to_date` to check and get atomically. There's currently only one case where we actually care if it was newly locked.